### PR TITLE
test(dtslint): add onErrorResumeNext

### PR DESCRIPTION
**Pull request by timdeschryver**
Description:
This PR adds dtslint tests for `onErrorResumeNext`.

Could it be that the following type is missing?

```ts
export function onErrorResumeNext<T, T2, R>(v: ObservableInput<T2>): OperatorFunction<T, R>;
```

Related issue (if exists): #4093

### CODE REVIEW

**Comment by cartant** [comment]
It's not an operator I've ever used. TBH, its signatures look wrong, to me. I think a lot of those signatures should return observables with a union type. E.g. `Observable<T, T2 | T3>`.

**Comment by timdeschryver** [comment]
I thought the same way while writing the tests.
I already had rewritten the types to see how it looked, what do you think?

**Comment by cartant** [comment]
LGTM.

I noticed that the existing typings (for `onErrorResumeNext` and `onErrorResumeNextStatic`) include a union in some of the signatures:

`...observables: Array<ObservableInput<any> | ((...values: Array<any>) => R)`

The `((...values: Array<any>) => R)` looks wrong to me. How can it make sense for a rest parameters type to be a non-array? I've had a quick look through the implementation and cannot see why it's needed. If you could double check that and remove it if you, too, think it should not be there, that would be great.

**Comment by timdeschryver** [comment]
At first sight, it seems like you're right.
The operators `combineLatest`, `zip` and `withLatestFrom` are also using the same signature tho...

Since I've changed the signatures, I should also update the compat signatures (https://github.com/ReactiveX/rxjs/blob/master/compat/operator/onErrorResumeNext.ts), right?

**Comment by cartant** [comment]
Those other operators have a result selector - ATM, at least - but `onErrorResumeNext` doesn't have one. It looks to me like a copy/paste mistake that happened once.

**Comment by timdeschryver** [comment]
I removed the type, I'll try to update the `compat` typings tomorrow or else during this week.

File: src/internal/operators/onErrorResumeNext.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ export function onErrorResumeNext<T, T2, T3, T4>(v: ObservableInput<T2>, v2: Obs
 export function onErrorResumeNext<T, T2, T3, T4, T5>(v: ObservableInput<T2>, v2: ObservableInput<T3>, v3: ObservableInput<T4>, v4: ObservableInput<T5>): OperatorFunction<T, T | T2 | T3 | T4 | T5>;
 export function onErrorResumeNext<T, T2, T3, T4, T5, T6>(v: ObservableInput<T2>, v2: ObservableInput<T3>, v3: ObservableInput<T4>, v4: ObservableInput<T5>, v5: ObservableInput<T6>): OperatorFunction<T, T | T2 | T3 | T4 | T5 | T6>;
 export function onErrorResumeNext<T, T2, T3, T4, T5, T6, T7>(v: ObservableInput<T2>, v2: ObservableInput<T3>, v3: ObservableInput<T4>, v4: ObservableInput<T5>, v5: ObservableInput<T6>, v6: ObservableInput<T7>): OperatorFunction<T, T | T2 | T3 | T4 | T5 | T6 | T7>;
-export function onErrorResumeNext<T, R>(...observables: Array<ObservableInput<any>>): OperatorFunction<T, R>;
-export function onErrorResumeNext<T, R>(array: ObservableInput<any>[]): OperatorFunction<T, R>;
+export function onErrorResumeNext<T, R>(...observables: Array<ObservableInput<any>>): OperatorFunction<T, T | R>;
```
Comment (timdeschryver) [inline src/internal/operators/onErrorResumeNext.ts]: Do you think this typing and the next one make sense?

**Comment by timdeschryver** [comment]
👍 , thanks for the update @cartant !

**Comment by cartant** [comment]
@benlesh FYI, I've reviewed and approved this, but I've not merged it, as it includes code (type signature) changes, that require your approval. But, IMO, it's good to go.

**Pull request merged**